### ISSUE TITLE
Termloop abruptly stops

### ISSUE THREAD

**Issue by JoelOtter** [JoelOtter/termloop#21]
Sometimes Termloop will simply stop, outputting no useful information, even if debugging is turned on.

If you've noticed this happening, please post your configuration as a comment. Here's mine:

**OS**: Mac OS X 10.10.5 (Yosemite)
**Go version**: 1.5
**Terminal**: iTerm2 + tmux

**Comment by JoelOtter** [comment]
I just managed to reproduce this by running the `collision.go` example and just holding the arrow keys in different directions until the game just stopped. Still no idea why it's happening.

**Comment by JoelOtter** [comment]
OK, narrowed down *slightly*. If I modify the `Game.Start()` method to look like this (adding two Printlns):

```go
// main function. By default, the escape key exits.
func (g *Game) Start() {
	// Init Termbox
	err := termbox.Init()
	termbox.SetOutputMode(termbox.Output256)
	termbox.SetInputMode(termbox.InputEsc | termbox.InputMouse)
	if err != nil {
		panic(err)
	}
	defer fmt.Println("END1")
	defer g.dumpLogs()
	defer termbox.Close()
	g.screen.resize(termbox.Size())

	// Init input
	g.input.start()
	defer g.input.stop()
	defer fmt.Println("END2")
	clock := time.Now()

mainloop:
	for {
... etc
```

...then "END1" is being printed but "END2" isn't. This means Termbox is still running until we kill it with `termbox.Close()` (because the defers are LIFO) but since "END1" is being printed this means all our defers are being processed If I put a print statement after the mainloop, though, that *doesn't* get printed, implying some kind of crash (I think). Will keep investigating...

**Comment by JoelOtter** [comment]
Found it.

```go
case ev := <-g.input.eventQ:
			if ev.Key == g.input.endKey {
				g.Log("endkey")
				break mainloop
			} else if EventType(ev.Type) == EventResize {
				g.screen.resize(ev.Width, ev.Height)
			} else if EventType(ev.Type) == EventError {
				g.Log(ev.Err.Error())
			}
			g.screen.Tick(convertEvent(ev))
```

When the game ends, it's because ev.Key is being read as KeyEsc (the escape key, which by default ends a game). This is because Termbox, by default, passes a KeyEsc if it doesn't recognise the key input.
Switching it to InputAlt mode - an unrecognised key is assumed to be a modifier and stored in a different place in the Event struct - seems to fix this. I'll do a bit more testing and then push this fix out. Thanks to @dylangleason for noticing this.

**Comment by dylangleason** [comment]
Thanks for addressing this. A bit late, but I reproduced this issue with the following config:

**OS**: Mac OS X 10.10.5 
**Go Version**: go1.5.1 darwin/amd64
**Terminal**: iTerm2

**Comment by dylangleason** [comment]
This issue is also confirmed using Terminal.app, version 2.5.3

**Comment by JoelOtter** [comment]
Also occurs in xterm under X11 (Xquartz)

**Comment by JoelOtter** [comment]
Just to update this after discussion on Gitter - changing Termbox's input mode to `InputAlt` solves this issue, but renders the Escape key useless, as it'll be read as the modifier on its own and not a key at all. Default `endKey` for Termloop will be changed to Ctrl+C. It'd be possible to keep the escape key working as an input, but I think it's better to unsupport it entirely, as it *will* cause unpredictable behaviour on OS X.

The chap who makes Termbox, nsf, has as pretty much his motto, "terminal input is really bad". I'm inclined to agree. :wink:

**Comment by JoelOtter** [comment]
This issue has been fixed in 1bec7cd. Default end key combination is now Ctrl+C.
